astarte/device/mapping.py

```diff
@@ -120,32 +143,40 @@ def __init__(self, mapping_definition: dict, interface_type: str):
             else:
                 self.reliability = 0
 
-    def validate(self, payload: MapType, timestamp: datetime) -> Tuple[bool, str]:
+    def validate(self, payload: MapType, timestamp: datetime) -> tuple[bool, str]:
         """
         Mapping data validation
+
         Parameters
         ----------
-        payload: object
+        payload: MapType
             Data to validate
         timestamp: datetime or None
             Timestamp associated to the payload
+
         Returns
         -------
-        success: bool
+        bool
             Success of the validation operation
-        msg: str
+        str
             Error message if success is False
         """
+        min_supported_int = -2147483648
+        max_supported_int = 2147483647
         # Check if the interface has explicit_timestamp when a timestamp is given (and viceversa)
         if self.explicit_timestamp and not timestamp:
             return False, f"Timestamp required for {self.endpoint}"
         if not self.explicit_timestamp and timestamp:
             return False, f"It's not possible to set the timestamp for {self.endpoint}"
         # Check the type of data is valid for that endpoint
         if not isinstance(payload, self.__actual_type):
-            return False, f"{self.endpoint} is {self.type} but {type(payload)} was provided"
-        # Must return False when trying to send an integer outside -2147483648 to 2147483647 interval.
-        if self.type == "integer" and not -2147483648 <= payload <= 2147483647:
+            return (
+                False,
+                f"{self.endpoint} is {self.type} but {type(payload)} was provided",
+            )
+        # Must return False when trying to send an integer outside -2147483648 to 2147483647
+        # interval.
```

Comment:
```suggestion
        # Must return False when trying to send an integer outside the allowed interval.
```
Not sure if the rephrasing is ok.
My suggestion is mostly to avoid hardcoded numbers in comments.
Same a couple of lines below.